libs/utils/src/lsn.rs

```diff
@@ -44,10 +44,10 @@ impl Lsn {
     /// Parse an LSN from a filename in the form `0000000000000000`
     pub fn from_filename<F>(filename: F) -> Result<Self, LsnParseError>
     where
-        F: AsRef<Path>,
+        F: AsRef<Utf8Path>,
     {
-        let filename: &Path = filename.as_ref();
-        let filename = filename.to_str().ok_or(LsnParseError)?;
+        let filename: &Utf8Path = filename.as_ref();
+        let filename = filename.as_std_path().to_str().ok_or(LsnParseError)?;
```

Comment:
There is https://docs.rs/camino/1.1.6/camino/struct.Utf8Path.html#method.as_str which seems applicable here instead of the fallible std::path::Path route?